filebeat/input/filestream/input_test.go

```diff
@@ -115,6 +116,49 @@ paths:
 	})
 }
 
+func TestTakeOverTags(t *testing.T) {
+
+	testCases := []struct {
+		name     string
+		takeOver bool
+		testFunc func(event beat.Event)
+	}{
+		{
+			name:     "test-take_over-true",
+			takeOver: true,
+			testFunc: func(event beat.Event) {
+				tags, err := event.GetValue("tags")
+				require.NoError(t, err)
+				require.Contains(t, tags, "take_over")
+			},
+		},
+		{
+			name:     "test-take_over-false",
+			takeOver: false,
+			testFunc: func(event beat.Event) {
+				_, err := event.GetValue("tags")
+				require.ErrorIs(t, err, mapstr.ErrKeyNotFound)
+			},
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			filename := generateFile(t, t.TempDir(), 5)
+			cfg := fmt.Sprintf(`
+type: filestream
+prospector.scanner.check_interval: 1s
+take_over: %t
+paths:
+    - %s`, testCase.takeOver, filename)
+			runner := createFilestreamTestRunner(context.Background(), t, testCase.name, cfg, 5, true)
+			events := runner(t)
+			for _, event := range events {
+				testCase.testFunc(event)
+			}
+		})
+	}
+}
```

Comment:
Assertions should be made with the `testing.T` from the test running. Those functions will be used in a sub test that will receive a different `testing.T` than the one you're capturing here. Methods like `t.Name` will return an 'incorrect' value.

```suggestion
func TestTakeOverTags(t *testing.T) {
	testCases := []struct {
		name     string
		takeOver bool
		testFunc func(t *testing.T, event beat.Event)
	}{
		{
			name:     "test-take_over-true",
			takeOver: true,
			testFunc: func(t *testing.T, event beat.Event) {
				tags, err := event.GetValue("tags")
				require.NoError(t, err)
				require.Contains(t, tags, "take_over")
			},
		},
		{
			name:     "test-take_over-false",
			takeOver: false,
			testFunc: func(t *testing.T, event beat.Event) {
				_, err := event.GetValue("tags")
				require.ErrorIs(t, err, mapstr.ErrKeyNotFound)
			},
		},
	}
	for _, testCase := range testCases {
		t.Run(testCase.name, func(t *testing.T) {
			filename := generateFile(t, t.TempDir(), 5)
			cfg := fmt.Sprintf(`
type: filestream
prospector.scanner.check_interval: 1s
take_over: %t
paths:
    - %s`, testCase.takeOver, filename)
			runner := createFilestreamTestRunner(context.Background(), t, testCase.name, cfg, 5, true)
			events := runner(t)
			for _, event := range events {
				testCase.testFunc(t, event)
			}
		})
	}
}
```

Reply:
I only changed the function signatures/calls, but GitHub does not seem to be able to do a diff here :man_shrugging:

Reply:
Ahh! Nice catch.
Thanks!